polaris-react/src/components/Button/Button.scss

```diff
@@ -767,94 +823,89 @@
   &.iconOnly {
     margin: calc(-1 * var(--p-space-2));
 
+    #{$se23} & > .Content::after {
+      top: -2px;
+      right: -2px;
+      bottom: -2px;
+      left: -2px;
+    }
+
     svg {
       fill: var(--p-color-icon);
+
+      #{$se23} & {
+        fill: var(--p-color-icon-interactive);
+      }
     }
 
     &:hover svg {
       fill: var(--p-color-icon-hover);
     }
 
+    #{$se23} &:hover,
+    #{$se23} &:focus:not(.disabled) {
+      fill: var(--p-color-icon-interactive-hover);
```

Comment:
I think these `iconOnly` + `plain` background colors should be transparent

<img width="814" alt="Screenshot 2023-06-08 at 12 10 24 PM" src="https://github.com/Shopify/polaris/assets/20652326/80089f29-2534-48fd-b335-41e027f4ba8c">

Reply:
It's hard for me to decipher the levels of nesting in the css to figure out which variant this is targeting but there are `plain` and `plain primary` variants. 

`plain` doesn't get the bg and `plain primary` does

Reply:
The style updates in this PR are scoped to `plain` and `plain destructive`.